### PR TITLE
Improve setup in README

### PR DESCRIPTION
**Pull request by jayeff**
* group steps into logical groups
* explain how to use vagrant as IoT Edge device
* explain how to use another device as IoT Edge device
* remove deployment.debug.template as it's not necessary

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,61 +4,75 @@ This is an [Azure IoT Edge](https://azure.microsoft.com/en-us/services/iot-edge/
...
 
-1. Checkout this repo `git clone https://github.com/crate/azure-iotedge-cratedb-sample.git`
-2. Install [Vagrant](https://www.vagrantup.com/downloads.html).
-3. Install [Visual Studio Code](https://code.visualstudio.com/download).
-4. Install extensions for Visual Studio Code: (On the Visual Studio Code: Menu -> Preferences -> Extensions)
+* Checkout this repo `git clone https://github.com/crate/azure-iotedge-cratedb-sample.git`
+* Install [Vagrant](https://www.vagrantup.com/downloads.html).
+* Install [Visual Studio Code](https://code.visualstudio.com/download).
+* Install extensions for Visual Studio Code: (On the Visual Studio Code: Menu -> Preferences -> Extensions)
     1. Azure IoT Hub Toolkit
```
Comment (kamejosh) [inline README.md]: i could only find "Azure IoT Hub" or "Azure IoT Tools"

File: README.md [hunk 2]
```diff
@@ -4,61 +4,75 @@ This is an [Azure IoT Edge](https://azure.microsoft.com/en-us/services/iot-edge/
...
-3. Open a web browser with `http://localhost:4200` and go to the CrateDB Admin UI console.
-4. Create the tables and the user by executing all SQL Statements in the file `./scripts/database_setup/createTable.sql`.
-5. Stop the container with `docker container stop cratedb`.
+* Install [Docker](https://www.docker.com/get-started).
 
+## Create the IoTHub and the IoT Edge device
 
-## Configuring the OPC-UA Publisher edge module
+* Create an IoT Hub (S1) on Azure. You'll need to wait for it to be in an active state, it could take a while.
+* After your IoT Hub has started, you can use Visual Studio Code integration to connect to it.
```
Comment (kamejosh) [inline README.md:27]: I was stuck here a little, maybe linking to this guide helps:
https://devblogs.microsoft.com/visualstudio/azure-iot-tools-help-you-connect-to-azure-iot-hub-in-1-minute-in-visual-studio-code/

File: README.md [hunk 3]
```diff
@@ -4,61 +4,75 @@ This is an [Azure IoT Edge](https://azure.microsoft.com/en-us/services/iot-edge/
...
+* In Visual Studio Code (or on the Azure Portal), you can create a new IoT Edge device.
+* Copy the contents of `.env.template` into a new file `.env`. The default settings work out of the box for the vagrant box. If you want to configure a different IoT Edge device please consult the 'Use a different device as IoT Edge Device' instructions.
+* Right click on `deployment.template.json` and select 'Build IoT Edge Solution'.
+* Go to the config folder, right click on `deployment.amd64.json` and select on 'Create Deployment for Single Device'.
+
+## Prefill CrateDB with required tables and users
+
+Before starting to deploy the sample to the edge device, the local CrateDB used in the sample has to be created first.
```
Comment (kamejosh) [inline README.md]: does this mean, the last step of the previous section should not have been done before this?

File: README.md [hunk 4]
```diff
@@ -4,61 +4,75 @@ This is an [Azure IoT Edge](https://azure.microsoft.com/en-us/services/iot-edge/
...
 
-This approach makes the set-up experience more streamlined.
+## Requirements
 
-1. Checkout this repo `git clone https://github.com/crate/azure-iotedge-cratedb-sample.git`
-2. Install [Vagrant](https://www.vagrantup.com/downloads.html).
-3. Install [Visual Studio Code](https://code.visualstudio.com/download).
-4. Install extensions for Visual Studio Code: (On the Visual Studio Code: Menu -> Preferences -> Extensions)
+* Checkout this repo `git clone https://github.com/crate/azure-iotedge-cratedb-sample.git`
+* Install [Vagrant](https://www.vagrantup.com/downloads.html).
```
Comment (kamejosh) [inline README.md]: vagrant seems to ned a 3rd party provider like, VBox, VMware or other, maybe mention that here?

File: README.md [hunk 5]
```diff
@@ -4,61 +4,75 @@ This is an [Azure IoT Edge](https://azure.microsoft.com/en-us/services/iot-edge/
...
+
+
+## Set up Vagrant VM as IoT Edge Device
 
-1. Per default this sample uses all nodes from the [OPC-UA Server simulation](https://github.com/Azure-Samples/iot-edge-opc-plc). If you want to change the setup, just edit the `publishednodes.json` file in the folder `appdata`
-2. Edit the files `deployment.template.json` and `deployment.debug.template.json` by changing the publisher's data folder from `C:/Dev/iotedge/crate/localstorageCrateDBEdgeSolution` to the directory you've downloaded the source repository on your local disk.
+* Right click on the device and select “Copy Device Connection String” from the Menu.
+* Modify the `vagrant/config.template.yaml` to use your Device Connection String and rename it to `config.yaml`.
+* In your terminal, start Vagrant by `vagrant up`.
+* Open the shh by running `vagrant ssh`.
```
Comment (kamejosh) [inline README.md:49]: this was unclear, i needed to open the terminal in the vagrant folder, create a new folder in de base directory called crate and then i could start vagrant with `vagrant up`

Comment (kamejosh) [reply README.md:49]: it was also confusing to get this error right at the end of `vagrant up`:
`default: A dependency job for iotedge.service failed. See 'journalctl -xe' for details.`
someone might think that it didn't work and will not connect via `vagrant ssh`

Comment (kamejosh) [reply README.md:49]: i got stuck here completely. vagrant was running and i could connect via ssh but crateDb was not reachable so the next section was not working. i will try again later but maybe you have some hints for me

**Review by kamejosh — COMMENTED** [review]
vagrant is causing headaches 😁

File: README.md (hunk 1 above)
Comment (jayeff) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (jayeff) [reply README.md]: not necessary to do beforehand. deleted this notice

File: README.md (hunk 4 above)
Comment (jayeff) [reply README.md]: Added VirtualBox

File: README.md (hunk 5 above)
Comment (jayeff) [reply README.md:49]: I think can be ignored as it works now.

**Review by kamejosh — APPROVED** [review]
LGTM

**Pull request merged**